src/components/lock/StakeIQ.tsx

```diff
@@ -96,12 +103,11 @@ const StakeIQ = ({ exchangeRate }: { exchangeRate: number }) => {
     }
   }, [data, toast, trxHash, checkPoint])
 
-  const updateIqToBeLocked = (value: string | number) => {
+  const updateIqToBeLocked = (value: Result | number | string) => {
```

Comment:
a function that accepts strings or result or number its a code smell that code is not very solid and might lead to unexpected results.

why there are multiple ways of update IQ w different formats? There should be only one to avoid having a bug sending a string vs sending a number vs sending a bigInt

Reply:
because the value can take be in a different format (Result format -when the user clicks the max button to max amount-, string format when the user inputs the lock value in the input field)...we have this complication because we don't want to convert the contract result directly to a number format. But I understand what you are saying.

Reply:
when user clicks max button it should take bigInt value that should be stored in the state
when user input, string should be converted to bigInt and sent to that function